File: src/main/java/io/github/sunshinewzy/shining/api/dictionary/IDictionaryItem.java
package io.github.sunshinewzy.shining.api.dictionary;

import io.github.sunshinewzy.shining.api.dictionary.behavior.ItemBehavior;
import io.github.sunshinewzy.shining.api.namespace.NamespacedId;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public interface IDictionaryItem {
	
	@NotNull
	NamespacedId getName();
	
	@NotNull
	ItemStack getItemStack();
	
	@NotNull
	List<ItemBehavior> getBehaviors();
	
	boolean hasName();
	
}

File: src/main/kotlin/io/github/sunshinewzy/shining/api/dictionary/IDictionaryItem.kt
package io.github.sunshinewzy.shining.api.dictionary

import io.github.sunshinewzy.shining.api.dictionary.behavior.ItemBehavior
import io.github.sunshinewzy.shining.api.namespace.NamespacedId
import org.bukkit.inventory.ItemStack

interface IDictionaryItem {
    
    fun getName(): NamespacedId
    
    fun getItemStack(): ItemStack
    
    fun getBehaviors(): List<ItemBehavior>
    
    fun hasName(): Boolean
    
}